lib/asset-manifest-inserter.js

```diff
@@ -0,0 +1,60 @@
+var BroccoliCachingWriter = require('broccoli-caching-writer');
+var path = require('path');
+var fs = require('fs-extra');
+var existsSync = require('exists-sync');
+var metaReplacer = require('./meta-handler').replacer;
+
+/**
+ * A Broccoli Plugin to modify index.html to include the asset manifest.
+ *
+ * @class AssetManifestInserter
+ * @extends BroccoliCachingWriter
+ *
+ * @private
+ * @param {Tree} tree
+ * @param {object} options
+ */
+
+AssetManifestInserter.prototype = Object.create(BroccoliCachingWriter.prototype);
+AssetManifestInserter.prototype.constructor = AssetManifestInserter;
+function AssetManifestInserter(inputNodes, options) {
+  options = options || {};
+  options.indexName = options.indexName || 'index.html';
+  options.cacheInclude = [options.indexName, 'tests/index.html', 'asset-manifest.json'];
+
+  BroccoliCachingWriter.call(this, inputNodes, {
+    annotation: options.annotation
+  });
+  this.options = options;
+}
+
+AssetManifestInserter.prototype.build = function() {
+  var sourceDir = this.inputPaths[0];
+  var manifestFilePath = path.join(sourceDir, 'asset-manifest.json');
```

Comment:
it seems like there are only 3 inputs that matter, yet the current usage will assume a rebuild is required if any of the inputs changes, also it will always taint the output even if the output has not changed.